app/questionnaire/value_source_resolver.py

```diff
@@ -217,6 +218,16 @@ def _resolve_location_source(self, value_source: Mapping) -> str | None:
     def _resolve_response_metadata_source(self, value_source: Mapping) -> str | None:
         return self.response_metadata.get(value_source.get("identifier"))
 
+    def resolve_list(self, value_source_list: list[dict]) -> list[ValueSourceTypes]:
```

Comment:
ah so recursive, if list, call itself for each item?
the only issue I could imagine is if there were lots of value sources in the transform arguments list all the isinstance checks might not be as fast but I suspect its fairly negligible 

I don't really mind, though since its a different arg type and different outcome, and easy to leave separate, I quite like it as it is